test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfiguration.cs

```diff
@@ -103,6 +105,32 @@ static object CreateExpectedModuleConfig(JObject source)
             return module;
         }
 
+        static object CreateExpectedAgentModuleConfig(JObject source)
+        {
+            JToken createOptions = source.SelectToken("settings.createOptions") ?? new JObject();
+            JToken env = source.SelectToken("env") ?? new JObject();
+
+            string createOptionsLabel = JsonConvert.SerializeObject(createOptions);
+
+            createOptions.Value<JObject>().TryAdd("Labels", new JObject());
+            JObject labels = createOptions.Value<JObject>("Labels");
+            labels.TryAdd("net.azure-devices.edge.create-options", new JValue(createOptionsLabel));
+            labels.TryAdd("net.azure-devices.edge.env", JsonConvert.SerializeObject(env));
+            labels.TryAdd("net.azure-devices.edge.owner", new JValue("Microsoft.Azure.Devices.Edge.Agent"));
+            createOptions["Labels"] = labels;
+            source["settings"]["createOptions"] = createOptions;
+            if (source["settings"] != null)
```

Comment:
`source["settings"]` will never be `null` here; you used it in the previous line to access `["createOptions"]`. 😉 

Since C# uses references you shouldn't need to update parent objects with pointers to their children. But it looks like my suggested code didn't handle the case where `source` doesn't have "settings" and/or "createOptions". a cleaner way to handle that is to use the TryAdd/Value pattern that you're already using for `labels`:

```csharp
            source.TryAdd("settings", new JObject());
            JObject settings = source.Value<JObject>("settings");

            settings.TryAdd("createOptions", new JObject());
            JObject createOptions = settings.Value<JObject>("createOptions");
            string createOptionsLabel = JsonConvert.SerializeObject(createOptions);

            createOptions.TryAdd("Labels", new JObject());
            JObject labels = createOptions.Value<JObject>("Labels");

            JToken env = source.SelectToken("env") ?? new JObject();

            labels.TryAdd("net.azure-devices.edge.create-options", new JValue(createOptionsLabel));
            labels.TryAdd("net.azure-devices.edge.env", JsonConvert.SerializeObject(env));
            labels.TryAdd("net.azure-devices.edge.owner", new JValue("Microsoft.Azure.Devices.Edge.Agent"));

            return CreateExpectedModuleConfig(source);
```